catroid/src/main/java/org/catrobat/catroid/ui/Multilingual.java

```diff
@@ -0,0 +1,469 @@
+					editor.commit();
+				}
+				if (position == 41) {
+					setLocale("ko");
+					SharedPreferences.Editor editor = languageSharedPreferences.edit();
+					editor.clear().apply();
+					editor.putString(languageTagKey, "ko");
+					editor.commit();
+				}
+				if (position == 42) {
+					setLocale("zh", "CN");
+					SharedPreferences.Editor editor = languageSharedPreferences.edit();
+					editor.clear().apply();
+					editor.putString(languageTagKey, "zh-rCN");
+					editor.commit();
+				}
+				if (position == 43) {
+					setLocale("zh", "TW");
+					SharedPreferences.Editor editor = languageSharedPreferences.edit();
+					editor.clear().apply();
+					editor.putString(languageTagKey, "zh-rTW");
+					editor.commit();
+				}
+				if (position == 44) {
+					setLocale("ar");
+					SharedPreferences.Editor editor = languageSharedPreferences.edit();
+					editor.clear().apply();
+					editor.putString(languageTagKey, "ar");
+					editor.commit();
+				}
+				if (position == 45) {
+					setLocale("ur");
+					SharedPreferences.Editor editor = languageSharedPreferences.edit();
+					editor.clear().apply();
+					editor.putString(languageTagKey, "ur");
+					editor.commit();
+				}
+				if (position == 46) {
+					setLocale("fa");
+					SharedPreferences.Editor editor = languageSharedPreferences.edit();
+					editor.clear().apply();
+					editor.putString(languageTagKey, "fa");
+					editor.commit();
+				}
+				if (position == 47) {
+					setLocale("ps");
+					SharedPreferences.Editor editor = languageSharedPreferences.edit();
+					editor.clear().apply();
+					editor.putString(languageTagKey, "ps");
+					editor.commit();
+				}
+				if (position == 48) {
+					setLocale("sd");
+					SharedPreferences.Editor editor = languageSharedPreferences.edit();
+					editor.clear().apply();
+					editor.putString(languageTagKey, "sd");
+					editor.commit();
+				}
+				if (position == 49) {
+					setLocale("iw");
+					SharedPreferences.Editor editor = languageSharedPreferences.edit();
+					editor.clear().apply();
+					editor.putString(languageTagKey, "iw");
+					editor.commit();
+				}
+			}
+		});
+	}
+
+	@TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
+	public static void setContextLocale(Context context, String languageTag) {
+		Locale language = new Locale(languageTag);
+		Resources resources = context.getResources();
+		DisplayMetrics displayMetrics = resources.getDisplayMetrics();
+		Configuration conf = resources.getConfiguration();
+		conf.locale = language;
+		Locale.setDefault(language);
+		conf.setLayoutDirection(language);
+		resources.updateConfiguration(conf, displayMetrics);
+	}
+
+	public void setLocale(String languageTag) {
+		setContextLocale(this, languageTag);
+		Intent intent = new Intent(Multilingual.this, MainMenuActivity.class);
+		startActivity(intent);
+		finishAffinity();
+	}
+
+	@TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
+	public static void setContextLocale(Context context, String languageTag, String countryTag) {
```

Comment:
I have to use both methods, one in Multilingual.java when name of the language is tapped to call the MainMenuActivity,
and the static method in CatrobatApplication.java after checking the specified shared preference value.